presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java

```diff
@@ -209,17 +208,23 @@ void abortOldTransactions()
     @VisibleForTesting
     void deleteOldShards()
     {
-        runTransaction(dbi, REPEATABLE_READ, (handle, status) -> {
+        try (Handle handle = dbi.open()) {
             ShardManagerDao dao = handle.attach(ShardManagerDao.class);
 
+            dao.dropTableTemporaryCreatedShards();
+            dao.createTableTemporaryCreatedShards();
+            dao.insertTemporaryCreatedShards();
             dao.insertDeletedShardsFromCreated();
             dao.deleteOldCreatedShards();
+            dao.dropTableTemporaryCreatedShards();
```

Comment:
We have to delete it up front in case it still exists, but it seems like good practice to clean it up at the end.